### ISSUE TITLE
Support underscore in project name

### ISSUE THREAD

**Issue by jacobsvante** [shuttle-hq/shuttle#262]
To harmonize with crate naming I think Shuttle supporting underscore in the project name as well would be reasonable. After all, underscore is a legal character in a domain name.

MRE:

```
cargo install cargo-shuttle@0.4.0
git clone git@github.com:shuttle-hq/shuttle.git
cd shuttle/examples/axum/hello-world
rm Shuttle.toml
sed -i.bkp 's/hello-world/hello_world/g' Cargo.toml
cargo shuttle run
```

Last line gives the following output:
```
Error:
`hello_world` is an invalid project name. project name must
1. not start or end with `-`.
2. not contain any characters outside of the alphanumeric range, except for `-`.
3. not be empty.
```

**Comment by coszio** [comment]
I can take this one

**Comment by chesedo** [comment]
Closes by #299
